packages/p2p/src/components/verification/verification.tsx

```diff
@@ -1,73 +1,101 @@
+            updated_url = `${updated_url}&${url_params.toString()}`;
+        }
+        window.location.href = updated_url;
+    };
     const checklist_items = [
         {
-            content: general_store.nickname || <Localize i18n_default_text='Choose your nickname' />,
-            status: general_store.nickname ? 'done' : 'action',
-            onClick: general_store.nickname
-                ? () => {
-                      //do nothing
-                  }
-                : () => {
-                      if (isDesktop()) showModal({ key: 'NicknameModal' });
-                      general_store.toggleNicknamePopup();
-                  },
-        },
-        {
-            content: general_store.poiStatusText(general_store.poi_status),
-            is_disabled: general_store.poi_status !== 'verified' && !general_store.nickname,
-            status: general_store.poi_status === 'verified' ? 'done' : 'action',
+            content: getPoiAction(poi_status),
+            is_disabled: poi_status === identity_status_codes.PENDING,
+            status: poi_status === identity_status_codes.VERIFIED ? 'done' : 'action',
             onClick:
-                general_store.poi_status === 'verified'
+                poi_status === identity_status_codes.VERIFIED
                     ? () => {
                           //do nothing
                       }
                     : () => {
-                          const search = window.location.search;
-                          let updated_url = `${routes.proof_of_identity}?ext_platform_url=${routes.cashier_p2p}`;
-                          if (search) {
-                              const url_params = new URLSearchParams(search);
-                              const updated_url_params = new URLSearchParams(updated_url);
-                              url_params.forEach((value, key) => updated_url_params.append(key, value));
-                              updated_url = `${updated_url}&${url_params.toString()}`;
-                          }
-                          window.location.href = updated_url;
+                          redirectToVerification(routes.proof_of_identity);
                       },
         },
+        ...(p2p_poa_required
+            ? [
+                  {
+                      content: getPoaAction(poa_status),
+                      is_disabled: poa_status === document_status_codes.PENDING,
+                      status: poa_status === document_status_codes.VERIFIED ? 'done' : 'action',
+                      onClick:
+                          poa_status === document_status_codes.VERIFIED
+                              ? () => {
+                                    //do nothing
+                                }
+                              : () => {
+                                    redirectToVerification(routes.proof_of_address);
+                                },
+                  },
```

Comment:
this looks the same as the one above for poi_status, can we just create a function that takes in the poi/poa status and passes it to each object element and return the entire object?